go-controller/pkg/config/config.go

```diff
@@ -1183,8 +1184,13 @@ func buildKubernetesConfig(exec kexec.Interface, cli, file *config, saPath strin
 	if Kubernetes.Kubeconfig != "" && !pathExists(Kubernetes.Kubeconfig) {
 		return fmt.Errorf("kubernetes kubeconfig file %q not found", Kubernetes.Kubeconfig)
 	}
-	if Kubernetes.CACert != "" && !pathExists(Kubernetes.CACert) {
-		return fmt.Errorf("kubernetes CA certificate file %q not found", Kubernetes.CACert)
+
+	if Kubernetes.CACert != "" {
```

Comment:
If there is no CACert and this is running in unprivleged mode, shouldn't we error? Or is that handled somewhere else?

Reply:
@trozet depends on whether the APIURL is HTTPS or plain HTTP; I suppose we could inspect that and fail early, but it'll fail very early anyway when cmd/ovnkube/ovnkube.go calls `NewOVNClientset()` without the CA data.

Reply:
@trozet updated the `newKubernetesRestConfig()` to validate that the CAData given and error out otherwise; which gets called very early everywhere.